lib/templates.ts

```diff
@@ -720,6 +725,260 @@ export async function updateIsPublishedForTemplate(
+      `Access revoked for ${userID}`
+    );
+
+    notifyOwnersOwnerRemoved(
+      userToRemove,
+      updatedTemplate.jsonConfig as FormProperties,
+      updatedTemplate.users
+    );
+
+    if (formCache.cacheAvailable) formCache.formID.invalidate(formID);
+  } catch (e) {
+    if (e instanceof AccessControlError)
+      logEvent(
+        ability.userID,
+        { type: "Form", id: formID },
+        "AccessDenied",
+        "Attempted to remove assigned user for form"
+      );
+    throw e;
+  }
+}
+
+/**
+ * Assign a user to a form
+ *
+ * @param ability
+ * @param formID
+ * @param userID
+ */
+export async function assignUserToTemplate(
+  ability: UserAbility,
+  formID: string,
+  userID: string
+): Promise<void> {
+  try {
+    const template = await getTemplateWithAssociatedUsers(ability, formID);
+
+    if (template === null) {
+      logMessage.warn(`Can not add user ${userID} to template ${formID}.  Template does not exist`);
+      throw new TemplateNotFoundError();
+    }
+
+    const userToAdd = await prisma.user.findUnique({
+      where: {
+        id: userID,
+      },
+      select: {
+        name: true,
+        email: true,
+      },
+    });
+
+    if (userToAdd === null) {
+      logMessage.warn(`Can not add user ${userID} to template ${formID}.  User does not exist`);
+      throw new UserNotFoundError();
+    }
+
+    const updatedTemplate = await prisma.template
+      .update({
+        where: {
+          id: formID,
+        },
+        select: {
+          jsonConfig: true,
+          users: true,
+        },
+        data: {
+          users: {
+            connect: {
+              id: userID,
+            },
+          },
+        },
+      })
+      .catch((e) => prismaErrors(e, null));
+
+    // No changes
+    if (updatedTemplate === null) return;
+
+    logEvent(
+      ability.userID,
+      { type: "Form", id: formID },
+      "GrantFormAccess",
+      `Access granted to ${userID}`
+    );
+
+    notifyOwnersOwnerAdded(
+      userToAdd,
+      updatedTemplate.jsonConfig as FormProperties,
+      updatedTemplate.users
+    );
+
+    if (formCache.cacheAvailable) formCache.formID.invalidate(formID);
+  } catch (e) {
+    if (e instanceof AccessControlError)
+      logEvent(
+        ability.userID,
+        { type: "Form", id: formID },
+        "AccessDenied",
+        "Attempted to remove assigned user for form"
+      );
+    throw e;
+  }
+}
+
+/**
+ * Notify all owners when ownership changes (owner added)
+ *
+ * @param user New owner
+ * @param form Form properties object
+ * @param users Current owners
+ */
+export const notifyOwnersOwnerAdded = async (
+  userToAdd: { name: string | null; email: string },
+  form: FormProperties,
+  users: { id: string; email: string }[]
+) => {
+  const emailContent = ownerAddedEmailTemplate(
+    form.titleEn,
+    form.titleFr,
+    userToAdd.name || userToAdd.email
+  );
+
+  users.forEach((owner) => {
+    sendEmail(owner.email, {
```

Comment:
Yeah, since the action will have completed at this point (the owner(s) have been added/removed), this would just be if the notification failed to send for some reason. I'm not sure displaying that information to the user will be useful. It will have been logged already by the notifyConnector.